### PR TITLE
846 inbound shipment service line mutations

### LINKED ISSUE #846: Inbound shipment service line mutations
Context:

I can create service charges in mSupply:

image

I can't in omSupply

Question:

Can we have mutations for inbound service lines

### PR DESCRIPTION
**Pull request by andreievg**
closes #846

Probably best to follow the commit for review. It's very much copy paste and replace PR

* Added test for outbound service line
* Copied and global replaced outbound to inbound service line service (added some more test data)
* Extracted normal inbound line into it's own folder
* Copied over outbound
* Copied and global replaced outbound to inbound service line graphql
* Copied and replace for batch mutations

### CODE REVIEW

File: graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/delete.rs [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+pub fn map_response(from: Result<String, ServiceError>) -> Result<DeleteResponse> {
+    let result = match from {
+        Ok(id) => DeleteResponse::Response(GenericDeleteResponse(id)),
+        Err(error) => DeleteResponse::Error(DeleteError {
+            error: map_error(error)?,
+        }),
+    };
+
+    Ok(result)
```
Comment (clemens-msupply) [inline graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/delete.rs:52]: I guess there is a reason why its pub and not lined? for later batch mutation?

File: service/src/invoice_line/inbound_shipment_service_line/update/validate.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    input: &UpdateInboundShipmentServiceLine,
+    connection: &StorageConnection,
+) -> Result<(InvoiceLineRow, InvoiceRow, ItemRow), UpdateInboundShipmentServiceLineError> {
+    let line = check_line_exists(&input.id, connection)?;
+    let invoice = check_invoice_exists(&input.invoice_id, connection)?;
+
+    let item = if let Some(item_id) = &input.item_id {
+        check_item(item_id, connection)?
+    } else {
+        check_item(&line.item_id, connection)?
```
Comment (clemens-msupply) [inline service/src/invoice_line/inbound_shipment_service_line/update/validate.rs:29]: this check_item is just for convenience? i.e. you just need to get the item also when it is not updated?

**Review by clemens-msupply — APPROVED** [review]

File: graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/delete.rs (hunk 1 above)
Comment (andreievg) [reply graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/delete.rs:52]: Yep, i did like the inline response conversion

File: service/src/invoice_line/inbound_shipment_service_line/update/validate.rs (hunk 2 above)
Comment (andreievg) [reply service/src/invoice_line/inbound_shipment_service_line/update/validate.rs:29]: It's copy and paste from outbound shipment, 2b26f92553ceec7f5d381df1f3234c6df90fb59c

**Comment by andreievg** [comment]
I've also updated similar to previous comment about destructuring input when inline_init is used 4f751984a358c967725d1fbb464389f2f718cf4a

**Pull request merged**